app/assets/javascripts/discourse/app/routes/tos.js

```diff
@@ -1,3 +1,36 @@
-import staticRouteBuilder from "discourse/lib/static-route-builder";
+import DiscourseRoute from "discourse/routes/discourse";
+import { inject as service } from "@ember/service";
+import DiscourseURL from "discourse/lib/url";
+import StaticPage from "discourse/models/static-page";
+import I18n from "I18n";
+import { action } from "@ember/object";
 
-export default staticRouteBuilder("tos");
+export default class TosRoute extends DiscourseRoute {
+  @service siteSettings;
+
+  activate() {
+    super.activate(...arguments);
+    DiscourseURL.jumpToElement(document.location.hash.slice(1));
+  }
+
+  beforeModel(transition) {
+    if (this.siteSettings["tos_url"]) {
```

Comment:
No big deal, but it feels a little weird to access site settings like this 🤷‍♂️ 
```suggestion
    if (this.siteSettings.tos_url) {
```